service/history/workflow/update/update.go

```diff
@@ -276,6 +276,7 @@ func (u *Update) WaitAccepted(ctx context.Context) (UpdateStatus, error) {
 func (u *Update) OnMessage(
 	ctx context.Context,
 	msg proto.Message,
+	isWorkflowRunning bool,
```

Comment:
Comments usually out of sync with the code because we update code but not the comments. It will be better to define a new type message which consist of two fields proto.Message and isWorkflowRunning. Then move onUpdate parsing of the message part to the new message class. This will leave OnMessage a pure double dispatch.